crates/bevy_math/src/aspect_ratio/mod.rs

```diff
@@ -0,0 +1,108 @@
+//! Provides a simple aspect ratio struct to help with calculations.
+
+pub mod error;
+
+use crate::{aspect_ratio::error::AspectRatioError, Vec2};
+
+#[cfg(feature = "bevy_reflect")]
+use bevy_reflect::Reflect;
+
+/// An `AspectRatio` is the ratio of width to height.
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd)]
+#[cfg_attr(feature = "bevy_reflect", derive(Reflect), reflect(Debug, PartialEq))]
+pub struct AspectRatio(f32);
+
+impl AspectRatio {
+    /// Standard 16:9 aspect ratio
+    pub const SIXTEEN_NINE: Self = Self(16.0 / 9.0);
+    /// Standard 4:3 aspect ratio
+    pub const FOUR_THREE: Self = Self(4.0 / 3.0);
+    /// Standard 21:9 ultrawide aspect ratio
+    pub const ULTRAWIDE: Self = Self(21.0 / 9.0);
+
+    /// Create a new [`AspectRatio`] from a given width and height.
+    ///
+    /// # Panics
+    /// Panics when the width and height values are invalid, such a zero or infinite widths/heights, or if either
+    /// value is `NaN`.
+    #[inline]
+    pub fn new(width: f32, height: f32) -> Self {
```

Comment:
IMO we should only provide the method currently called `try_new`: users can unwrap on their own. This shouldn't be done a lot, so I don't mind the increased burden.